reflex/experimental/assets.py

```diff
@@ -1,57 +1,90 @@
 """Helper functions for adding assets to the app."""
 
 import inspect
+import os
 from pathlib import Path
 from typing import Optional
 
 from reflex import constants
 
 
-def asset(relative_filename: str, subfolder: Optional[str] = None) -> str:
-    """Add an asset to the app.
+def asset(
+    path: str,
+    shared: bool = False,
+    subfolder: Optional[str] = None,
+) -> str:
+    """Add an asset to the app, either shared as a symlink or local.
+
+    Shared/External/Library assets:
     Place the file next to your including python file.
-    Copies the file to the app's external assets directory.
+    Links the file to the app's external assets directory.
 
     Example:
     ```python
-    rx.script(src=rx._x.asset("my_custom_javascript.js"))
-    rx.image(src=rx._x.asset("test_image.png","subfolder"))
+    # my_custom_javascript.js is a shared asset located next to the including python file.
+    rx.script(src=rx._x.asset(path="my_custom_javascript.js", shared=True))
+    rx.image(src=rx._x.asset(path="test_image.png", shared=True, subfolder="subfolder"))
+    ```
+
+    Local/Internal assets:
+    Place the file in the app's assets/ directory.
+
+    Example:
+    ```python
+    # local_image.png is an asset located in the app's assets/ directory. It cannot be shared when developing a library.
+    rx.image(src=rx._x.asset(path="local_image.png"))
     ```
 
     Args:
-        relative_filename: The relative filename of the asset.
-        subfolder: The directory to place the asset in.
+        path: The relative path of the asset.
+        subfolder: The directory to place the shared asset in.
+        shared: Whether to expose the asset to other apps.
 
     Raises:
         FileNotFoundError: If the file does not exist.
+        ValueError: If subfolder is provided for local assets.
 
     Returns:
-        The relative URL to the copied asset.
+        The relative URL to the asset.
     """
+    assets = constants.Dirs.APP_ASSETS
+    backend_only = os.environ.get(constants.ENV_BACKEND_ONLY)
```

Comment:
@benedikt-bartscher hey Bene, I found a minor bug - here you will get either a string or a None which means when you are checking for 

```python

if not backend_only:
   ...
 ```
if the env var is "False", it will be evaluated as True

so I suggest doing sth like:

```python

backend_only = os.environ.get(constants.ENV_BACKEND_ONLY, "false").lower() == "true"

```

Reply:
@maximvlah Thank you for analyzing it down to the root.